moveit_core/robot_state/test/test_aabb.cpp

```diff
@@ -151,32 +151,34 @@ TEST_F(TestAABB, TestPR2)
   // Initialize a ROS publisher
   char* argv[0];
   int argc = 0;
-  ros::init(argc, argv, "visualize_pr2");
-  ros::NodeHandle nh;
-  ros::Publisher pub_aabb = nh.advertise<visualization_msgs::Marker>("/visualization_aabb", 10);
-  ros::Publisher pub_obb = nh.advertise<visualization_msgs::Marker>("/visualization_obb", 10);
+  rclcpp::init(argc, argv);
+  auto node = rclcpp::Node::make_shared("visualize_pr2");
+  auto pub_aabb = node->create_publisher<visualization_msgs::Marker>("/visualization_aabb", rmw_qos_profile_default);
+  auto pub_obb = node->create_publisher<visualization_msgs::Marker>("/visualization_obb", rmw_qos_profile_default);
+  rclcpp::Rate loop_rate(10);
 
   // Wait for the publishers to establish connections
   sleep(5);
 
   // Prepare the ROS message we will reuse throughout the rest of the function.
-  visualization_msgs::Marker msg;
-  msg.header.frame_id = pr2_state.getRobotModel()->getRootLinkName();
-  msg.type = visualization_msgs::Marker::CUBE;
-  msg.color.a = 0.5;
-  msg.lifetime.sec = 3000;
+  auto msg = std::make_shared<visualization_msgs::Marker>();
```

Comment:
I think it's perfectly fine to use a reference instead of a shared pointer here.